statsrelay.go

```diff
@@ -88,12 +97,33 @@ func getSockBufferMaxSize() (int, error) {
 // the metric key name and returns it as a string.
 func getMetricName(metric []byte) (string, error) {
 	// statsd metrics are of the form:
-	//    KEY:VALUE|TYPE|RATE
+	//    KEY:VALUE|TYPE|RATE or KEY:VALUE|TYPE|RATE|#tags
+	var metricName string
 	length := bytes.IndexByte(metric, byte(':'))
 	if length == -1 {
 		return "error", errors.New("Length of -1, must be invalid StatsD data")
 	}
-	return string(metric[:length]), nil
+	if len(metricsPrefix) != 0 {
+		metricName = genPrefix(metric[:length], metricsPrefix)
+	} else {
+		metricName = string(metric[:length])
+	}
+	return metricName, nil
+}
+
+func genPrefix(metric []byte, metricsPrefix string) string {
+	return metricsPrefix + "." + string(metric)
+}
+
+func addPrefix(metric []byte, metricsPrefix string) []byte {
+	// statsd metrics are of the form:
+	//    KEY:VALUE|TYPE|RATE or KEY:VALUE|TYPE|RATE|#tags
+	length := bytes.IndexByte(metric, byte(':'))
+	if length == -1 {
+		return []byte("error")
```

Comment:
You should return an error type here rather than potentially putting non-statsd format strings in the statsd data.  Similar to `getMetricName`.